app/packages/core/src/plugins/SchemaIO/components/DynamicIO.tsx

```diff
@@ -19,9 +19,10 @@ export default function DynamicIO(props) {
   // todo: need to improve initializing default value in state
   useEffect(() => {
     if (
-      isPrimitiveType(type) &&
-      data !== defaultValue &&
-      !isPathUserChanged(path)
+      !isCompositeView(schema) &&
```

Comment:
IIUC composite views should not call onChange because their children will?

Reply:
Yah, that's right. At the moment, the initialization of default value provided by resolved schema is done through onChange on component mount (I do want to optimize this in future)